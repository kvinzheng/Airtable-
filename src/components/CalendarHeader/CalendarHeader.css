.CalendarHeader {
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  margin: 1rem 0 1.1rem 0;
}

.CalendarHeader-button span {
  display: inline-block;
  padding: 12px
}
.CalendarHeader-create-icon {
  display:inline-block;
  margin: 10px 0px 10px 10px;
}
.CalendarHeader-button {
  margin: 0 0.2rem 0 0.2rem;
  padding: 0;
  overflow: hidden;
  border-width: 0;
  outline: none;
  border-radius: 20px;
  background-color: white;
  color: #3c4043;;
  font-weight: bold;
  border: 1px grey solid;
  transition: background-color 0.3s;
}
.CalendarHeader-button-add {
  display: flex;
  justify-content: center;
  align-items: center;
}

.CalendarHeader-button:hover,
.CalendarHeader-button:focus {
  background-color: #D3D3D3;
  cursor: pointer;
}

.CalendarHeader-button-today {
  background-color: #359ee1;
}

.CalendarHeader-title {
  font-size: 22px;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 28px;  
  color: #3c4043;;
  text-align: left;
  display: inline-block;
  margin-left: 20px;
}

.CalendarEventList-content {
  display: flex;
  flex-direction: column;
  height: 90%;
  width: 100%;
}

.CalendarHeader-zoom:hover{ 
  cursor: pointer;
  transform: scale(1.5);

}
.CalendarHeader-zoom{
  margin: 0 5px;
  transition: transform .2s; /* Animation */
}

.CalendarHeader-right-portion {
  display: flex;
  justify-content: center;
  align-items: center;
}