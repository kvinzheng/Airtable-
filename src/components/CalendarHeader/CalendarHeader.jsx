import React from "react";
import { getNextMonth, formatMonthYear } from "../../helpers/dateFormat.js";
import { AiOutlineZoomIn, AiOutlineZoomOut } from "react-icons/ai";
import { FaPlus } from "react-icons/fa";
import { connect } from "react-redux";
import {
  setEvents,
  setZoomWidth,
  setEditableEvent,
} from "../../actions/events";
import { setCalendarForm } from "../../actions/form";
import PropTypes from "prop-types";

import "./CalendarHeader.css";

export const CalendarHeader = ({
  displayedDate,
  onMonthShift,
  zoomWidth,
  setCalendarForm,
  setZoomWidth,
  setEditableEvent,
}) => {
  const handleFormCreate = () => {
    setEditableEvent(null);
    setCalendarForm(true);
  };
  const handleZoomIn = () => {
    setZoomWidth(zoomWidth + 10);
  };
  const handleZoomOut = () => {
    if (zoomWidth > 90) {
      setZoomWidth(zoomWidth - 10);
    } else {
      alert("Zoom reach minimum");
    }
  };
  const handleZoomReset = () => {
    setZoomWidth(150);
  };
  return (
    <div className="CalendarHeader">
      <div className="CalendarHeader-navigation">
        <button
          title={"today"}
          className="CalendarHeader-button CalendarHeader-today"
          type="button"
          data-testid="CalendarHeader-today"
          onClick={() => onMonthShift(new Date())}
        >
          <span aria-hidden="true">Today</span>
        </button>
        <button
          title={"previous month"}
          className="CalendarHeader-button"
          type="button"
          onClick={() => onMonthShift(getNextMonth(displayedDate, -1))}
        >
          <span aria-hidden="true">{"<"}</span>
        </button>

        <button
          className="CalendarHeader-button"
          type="button"
          title={"next month"}
          onClick={() => onMonthShift(getNextMonth(displayedDate, +1))}
        >
          <span aria-hidden="true">{">"}</span>
        </button>
        <div className="CalendarHeader-title" data-testid="CalendarHeader-date">
          {formatMonthYear(displayedDate)}
        </div>
      </div>

      <div className="CalendarHeader-right-portion">
        <AiOutlineZoomIn
          disabled={zoomWidth < 90}
          onClick={handleZoomIn}
          className="CalendarHeader-zoom"
          data-testid="CalendarHeader-zoom-in"
          size={30}
          title="zoom in"
        />

        <AiOutlineZoomOut
          onClick={handleZoomOut}
          className="CalendarHeader-zoom"
          data-testid="CalendarHeader-zoom-out"
          size={30}
          title="zoom out"
        />

        <button
          tabIndex={-1}
          className="CalendarHeader-button"
          data-testid="CalendarHeader-zoom-reset"
          onClick={handleZoomReset}
        >
          <span>onZoomReset</span>
        </button>

        <button
          className="CalendarHeader-button CalendarHeader-button-add"
          data-testid="CalendarHeader-form-create"
          type="button"
          onClick={handleFormCreate}
        >
          <FaPlus size={20} className="CalendarHeader-create-icon" />
          <span> Create</span>
        </button>
      </div>
    </div>
  );
};

export const mapStateToProps = (state) => {
  return {
    form: state.form.calendarForm,
    zoomWidth: state.events.zoom.width,
  };
};

CalendarHeader.propTypes = {
  form: PropTypes.bool,
  zoomWidth: PropTypes.number,
};

export default connect(mapStateToProps, {
  setEvents,
  setCalendarForm,
  setZoomWidth,
  setEditableEvent,
})(CalendarHeader);
