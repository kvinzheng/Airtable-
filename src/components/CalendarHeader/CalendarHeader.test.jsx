import React from "react";
import thunk from "redux-thunk";
import { Provider } from "react-redux";

import Adapter from "enzyme-adapter-react-16";
import configureMockStore from "redux-mock-store";

import { shallow, configure } from "enzyme";
import { render, screen, fireEvent } from "@testing-library/react";

import { CalendarHeader } from "./CalendarHeader";
import { mapStateToProps } from "./CalendarHeader";
import MockData from "../../helpers/testingData";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  events: { data: [], zoom: { width: 150 } },
  form: { calendarForm: false },
});

configure({ adapter: new Adapter() });

describe("CalendarHeader component", () => {
  it("should render without any data defined", () => {
    const component = shallow(<CalendarHeader events={[]} form={false} />);

    expect(component).toMatchSnapshot();
  });

  it("should render with props passed in", () => {
    const component = shallow(
      <CalendarHeader
        zoomWidth={150}
        form={false}
        displayedDate={new Date()}
        onMonthShift={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("mapStateToProps", () => {
    const state = {
      events: { data: [], zoom: { width: 150 } },
      form: { calendarForm: false },
    };
    const expected = {
      zoomWidth: 150,
      form: false,
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it("When you click on multiple buttons on the header", () => {
    const setEditableEvent = jest.fn();
    const setCalendarForm = jest.fn();
    const setZoomWidth = jest.fn();
    const onMonthShift = jest.fn();

    render(
      <CalendarHeader
        setEditableEvent={setEditableEvent}
        zoomWidth={MockData.events.zoom.width}
        setEditableEvent={setEditableEvent}
        setCalendarForm={setCalendarForm}
        setZoomWidth={setZoomWidth}
        onMonthShift={onMonthShift}
      />
    );
    fireEvent.click(screen.getAllByTestId("CalendarHeader-zoom-in")[0]);
    fireEvent.click(screen.getAllByTestId("CalendarHeader-zoom-out")[0]);
    fireEvent.click(screen.getAllByTestId("CalendarHeader-zoom-reset")[0]);
    fireEvent.click(screen.getAllByTestId("CalendarHeader-form-create")[0]);
    expect(setEditableEvent).toHaveBeenCalled();
    expect(setCalendarForm).toHaveBeenCalled();
    expect(setZoomWidth).toHaveBeenCalled();
  });
});
