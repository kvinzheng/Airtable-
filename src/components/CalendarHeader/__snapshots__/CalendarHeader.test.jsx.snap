// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader component should render with props passed in 1`] = `ShallowWrapper {}`;

exports[`CalendarHeader component should render without any data defined 1`] = `ShallowWrapper {}`;
